app/controllers/reviews_controller.rb

```diff
@@ -28,24 +29,40 @@ def create
     end
   end
 
+  def edit
+    @release = Release.find(params[:release_id])
+    @review = Review.find(params[:id])
+    @release_genres = @release.genres
+  end
+
   def update
     if current_user
       @review = Review.find(params[:id])
-      @release = Release.find(params[:release_id])
-      if params[:commit] == "Upvote"
-        @review.votes += 1
-        @review.save
-      elsif params[:commit] == "Downvote"
-        @review.votes -= 1
-        @review.save
+      release = Release.find(@review.release)
+      if @review.update(review_params)
+        flash[:notice] = "Review successfully updated"
+      else
+        render :edit
```

Comment:
Hey Lily, I fixed this. I removed the logic for current_user since it was old, and the code on the view page was refactored to show the 'Edit' link only if they made the post. Also we're using the active record errors on our form that will show up once the page re-renders.